### PR TITLE
Topography Documentation

### PR DESCRIPTION
**Pull request by rosscastle**
First draft with updated index and link in idealized moist phys docs. This is quite rough as I know there are changes to make, all suggestions and additions very welcome.

Specifically in Section 2, I realised I have only used `land_generator_fn.py` once (I got confused with Ruths land generator code). What else should we include here? 

Should we talk about Gaussian or smoothed topography? 

What else have I missed?

I need make some changed in the `topography.f90` file before this should become live, I have written the docs as if I already have. I am editing `topography.f90` to accept any input file, it is currently hardcoded to one file. I should get to this next week.

### CODE REVIEW

File: docs/source/modules/topography.rst [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+There are other subroutines called by the above. These are listed below:
+``open_topog_file``, ``interp_topog``, ``find_indices``, ``input_data``, ``interp_water``, ``determine_ocean_points``, ``read_namelist``.
+
+References
+----------
+None needed at present.
+   
+Authors
+-------
+This documentation was written by Ross Castle, peer reviewed by Ruth Geen, and quality controlled by Marrianne Pietschnig.
```
Comment (mp586) [inline docs/source/modules/topography.rst:136]: Marianne* please 😁

File: docs/source/modules/topography.rst [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+- No topography - the default.
+- Flat topography - the surface geopotential is 0.
+- Interpolated topography - where it will call the ``topography`` module below, however this set up is not currently used.
+- Gaussian Topography - Simple Gaussian-shaped mountains are generated from specified parameters.
+
+2 - Creating Custom Topography
+------------------------------
+
+We provide python code that allows land files to be generated for a range of scenarios. This code is found in ``src/extra/python/isca/land_generator_fn.py``. Some tweaking of the code will be needed to suit the users requirements. The most useful feature of this code is interpolating high resolution data into the correct spectral grid size.
+
```
Comment (mp586) [inline docs/source/modules/topography.rst:63]: I guess the last sentence ("The most useful feature ... ") belongs to Section 3? I don't think there is any interpolation from high-res landmasks to other grids in land_generator_fn.py

File: docs/source/modules/topography.rst [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+As mentioned above, this module is generally not called anymore, in the normal Isca set up, land masks are actually specified through the ``idealized_moist_phys`` module, and the model topography through the ``spectral_init_cond`` module. The main use of it is to provide the "subgrid topography" when using the orographic gravity wave drag scheme (``mg_drag``).
+
+The fields that can be generated with this module are mean and standard deviation of topography within the specified grid boxes; and land-ocean (or water) mask and fractional area. The interpolation conserves the area weighted average of the input data by using the ``horiz_interp`` module.
+
+Namelist options
+^^^^^^^^^^^^^^^^
+| ``topog_file`` - The topography file that you wish to use.
+| ``water_file`` - The water data file (not commonly used, not provided in the Isca release)
+| ``file_in_rad`` - Is the topography file axis measured in radians? (Default False)
+| ``file_in_rad`` - Is the topography file axis measured in degrees? (Default True)
```
Comment (mp586) [inline docs/source/modules/topography.rst:93]: file_in_rad twice

File: docs/source/modules/topography.rst [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+There are no diagnostics available directly through this module. The subgrid topography variance can be obtained through damping driver by asking for ``sgsmtn`` (``mg_drag`` must be turned on).
+
+Relevant subroutines
+^^^^^^^^^^^^^^^^^^^^
+
+NOTE: Some subroutines have dimensional variants, e.g. interp_topog has both a 1d and 2d variant. 
+
+**Public Subroutines**
+
+These subroutines are used by the topography module to produce the land-masks etc that are being asked for by the user. They are largely self explainitory.
```
Comment (mp586) [inline docs/source/modules/topography.rst:115]: self-explanatory

File: docs/source/modules/topography.rst [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+| ``get_topog_mean`` returns the mean height from a region of the topography file so that that value can be used as the value when interpolating onto a smaller grid.
+| ``get_topog_stdev`` returns the standard deviation from a region of the topography file so that that value can be assoiated with the same region when interpolating onto a smaller grid.
+| ``get_ocean_frac`` returns the fraction of the land mask that is covered by ocean.
+| ``get_ocean_mask`` returns an ocean/land mask
+| ``get_water_frac`` returns the fraction of the land mask that is covered by water.
+| ``get_water_mask`` returns a water/land mask
```
Comment (mp586) [inline docs/source/modules/topography.rst:122]: I'm a bit confused by this, sorry: is there a different type of water available in Isca, like lakes/rivers? or is water_frac = ocean_frac?

**Review by mp586 — COMMENTED** [review]
Looks good to me except for some minor typos and stuff :)

**Comment by RuthG** [comment]
Hi Ross,

I've had a quick look over just now, looks good overall. A few minor notes:
- Perhaps should give units for topography height in the input file (m I believe)
- 'Can I have land without topography' - worth noting that it is also possible to generate aquamountains (topography without land)

Typos:
- "Topography is a seperate field" - should be separate
- 'Where there is ocean/land as so" - I think one of as or so should be deleted
- Summary section: 'latitutde-longitude' - should be latitude-longitude

On the 1st line of the idealised moist phys section the link to the other docs displays oddly for me, but that might be because I just viewed it through Github's browser rather than properly launching the website.

**Review by RuthG — COMMENTED** [review]
Hi Ross,

I've had a quick look over just now, looks good overall. A few minor notes:

Perhaps should give units for topography height in the input file (m I believe)
'Can I have land without topography' - worth noting that it is also possible to generate aquamountains (topography without land)
Typos:

"Topography is a seperate field" - should be separate
'Where there is ocean/land as so" - I think one of as or so should be deleted
Summary section: 'latitutde-longitude' - should be latitude-longitude
On the 1st line of the idealised moist phys section the link to the other docs displays oddly for me, but that might be because I just viewed it through Github's browser rather than properly launching the website.

**Review by RuthG — APPROVED** [review]
Typos from last review all sorted, looks fine to me

**Pull request merged**